tests/components/worldclock/test_sensor.py

```diff
@@ -15,15 +15,35 @@ def setUp(self):
         self.hass = get_test_home_assistant()
         self.time_zone = dt_util.get_time_zone("America/New_York")
 
+    def test_time(self):
+        """Test the time at a different location."""
         config = {"sensor": {"platform": "worldclock", "time_zone": "America/New_York"}}
-
         assert setup_component(self.hass, "sensor", config)
         self.hass.block_till_done()
         self.addCleanup(self.hass.stop)
 
-    def test_time(self):
-        """Test the time at a different location."""
         state = self.hass.states.get("sensor.worldclock_sensor")
         assert state is not None
 
         assert state.state == dt_util.now(time_zone=self.time_zone).strftime("%H:%M")
+
+    def test_time_format(self):
+        """Test time_format setting."""
+        time_format = "%a, %b %d, %Y %I:%M %p"
+        config = {
+            "sensor": {
+                "platform": "worldclock",
+                "time_zone": "America/New_York",
+                "time_format": time_format,
+            }
+        }
+        assert setup_component(self.hass, "sensor", config)
+        self.hass.block_till_done()
+        self.addCleanup(self.hass.stop)
+
+        state = self.hass.states.get("sensor.worldclock_sensor")
+        assert state is not None
+
+        assert state.state == dt_util.now(time_zone=self.time_zone).strftime(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)